AirCasting/Graph/External/ExternalGraph.swift

```diff
@@ -0,0 +1,168 @@
+import SwiftUI
+import Charts
+import Resolver
+
+struct ExternalGraph: UIViewRepresentable {
+    typealias UIViewType = AirCastingGraph
+    typealias OnChange = (ClosedRange<Date>) -> Void
+    
+    @InjectedObject private var userSettings: UserSettings
+    @ObservedObject var stream: MeasurementStreamEntity
+    @ObservedObject var thresholds: SensorThreshold
+    private var rangeChangeAction: OnChange?
+    
+    var isAutozoomEnabled: Bool
+    let simplifiedGraphEntryThreshold = 1000
+    
+    init(stream: MeasurementStreamEntity, thresholds: SensorThreshold, isAutozoomEnabled: Bool) {
+        self.stream = stream
+        self.thresholds = thresholds
+        self.isAutozoomEnabled = isAutozoomEnabled
+    }
+    
+    func onDateRangeChange(perform action: @escaping OnChange) -> Self {
+        var newGraph = self
+        newGraph.rangeChangeAction = action
+        return newGraph
+    }
+    
+    func makeUIView(context: Context) -> AirCastingGraph {
+        let uiView = AirCastingGraph(onDateRangeChange: { newRange in
+            rangeChangeAction?(newRange)
+        })
+        try? uiView.updateWithThreshold(thresholdValues: thresholds.rawThresholdsBinding.wrappedValue)
+
+        let entries = stream.allMeasurements?.sorted(by: { $0.time < $1.time }).compactMap({ measurement -> ChartDataEntry? in
```

Comment:
You can also use `.sorted { $0.time < $1.time }` a little bit shorter version. Also in the line 71.

Reply:
I would stay with `by` prefix as it sound better to me as long as it cool with you?

Reply:
I would stay with `by` prefix as it sound better to me as long as it cool with you?

Reply:
Of course, it was just a suggestion.